src/javaharness/java/arcs/android/impl/LocalArcsImpl.java

```diff
@@ -0,0 +1,53 @@
+package arcs.android.impl;
+
+import javax.inject.Inject;
+
+import arcs.api.Arc;
+import arcs.api.ArcsEnvironment;
+import arcs.api.ArcsImpl;
+import arcs.api.PECInnerPort;
+import arcs.api.PecPortManager;
+import arcs.api.PortableJson;
+
+public class LocalArcsImpl extends ArcsImpl {
+
+  @Inject protected ArcsEnvironment environment;
+  @Inject protected PecPortManager pecPortManager;
```

Comment:
Constructor injection is generally preferable to field-level injection. This would be better:

```
private final ArcsEnvironment environment;
private final PecPortManager pecPortManager;
@Inject
LocalArcsImpl(
    ArcsEnvironment environment,
    PecPortManager pecPortManager,
    PortableJsonParser jsonParser,
    UiBroker uiBroker) {
  super(jsonParser, uiBroker);
}
```

And similarly, remove `@Inject` from the members of the ArcsImpl base class, make those members `protected final`, and add a constructor that sets them (the constructor can't be injected though because it's an abstract class).

Reply:
done